src/utils/get-code-from-ast.ts

```diff
@@ -9,23 +9,32 @@ import { removeNodesFromOriginalCode } from './remove-nodes-from-original-code';
  * This function generate a code string from the passed nodes.
  * @param nodes All imports, in the sorted order in which they should appear in
  * the generated code.
+ * @param importNodes All nodes that were originally relevant. (This includes nodes that need to be deleted!)
  * @param originalCode The original input code that was passed to this plugin.
  * @param directives All directive prologues from the original code (e.g.
  * `"use strict";`).
  * @param interpreter Optional interpreter directives, if present (e.g.
  * `#!/bin/node`).
  */
-export const getCodeFromAst = (
-    nodes: Statement[],
-    originalCode: string,
-    directives: Directive[],
-    interpreter?: InterpreterDirective | null,
-) => {
+export const getCodeFromAst = ({
+    nodes,
+    importNodes = nodes,
```

Comment:
Yes, `importNodes` is a superset, we need to pass the original nodes so that the old code-locations can be deleted when we merge a node. Once I was deleting some of the nodes from the `nodes`-array then the generated output code had the properly sorted & modified nodes followed by the unmodified nodes I wanted to have deleted.

How do you feel about these names?

```ts
/**
 * This function generates a code string from the passed nodes.
 * @param nodesToOutput The remaining imports which should be rendered. (Node specifiers & types may be mutated)
 * @param allOriginalImportNodes All import nodes that were originally relevant. (This includes nodes that need to be deleted!)
 * @param originalCode The original input code that was passed to this plugin.
 * @param directives All directive prologues from the original code (e.g.
 * `"use strict";`).
 * @param interpreter Optional interpreter directives, if present (e.g.
 * `#!/bin/node`).
 */
export const getCodeFromAst = ({
    nodesToOutput,
    allOriginalImportNodes = nodesToOutput,
    originalCode,
    directives,
    interpreter,
}: {
    nodesToOutput: Statement[];
    allOriginalImportNodes?: Statement[];
    originalCode: string;
    directives: Directive[];
    interpreter?: InterpreterDirective | null;
}) => {
```